test/features.js

```diff
@@ -0,0 +1,292 @@
+import {setup, teardown, shouldThrowError} from './utils';
+
+let db;
+
+describe('Security tests', () => {
+  beforeEach(() => {
+    db = setup();
+  });
+  afterEach(teardown);
+  it('get when unset', async () => {
+    (await db.getSecurity()).should.eql({});
+  });
+  it('put', async () => {
+    const secObj = {members: {roles: 'test'}};
+    (await db.putSecurity(secObj)).should.eql({ok: true});
+    (await db.getSecurity()).should.eql(secObj);
+  });
+});
+
+describe('Installed security tests', () => {
+  const rightlessUser = {name: null, roles: []};
+  beforeEach(async () => {
+    db = setup();
+    db.installSecurityMethods();
+    await db.putSecurity({
+      admins: {
+        names: ['admin'],
+        roles: ['admin']
+      },
+      members: {
+        names: ['member'],
+        roles: ['member']
+      }
+    })
+  });
+  afterEach(async () => {
+    db.uninstallSecurityMethods()
+    await teardown();
+  });
+
+  it('basic', () => {
+    // lets setUp() and tearDown() do all the work...
+  })
+  it('double install', () => {
+    //1 install: setUp()
+		//2 install:
+    (()=> db.installSecurityMethods()).should.throw(Error);
+  });
+  it('double uninstall', () => {
+    //1 remove:
+		db.uninstallSecurityMethods();
+		//2 remove:
+		(() => db.uninstallSecurityMethods()).should.throw(Error);
+		//recover for tearDown()
+		db.installSecurityMethods();
+  });
+  it('all docs', async () => {
+    const resp = await db.allDocs({userCtx: {
+			name: 'admin',
+			roles: []
+		}});
+		resp.total_rows.should.equal(0);
+
+    const err = await shouldThrowError(async () => {
+      await db.allDocs({userCtx: {
+				'name': 'unknown',
+				'roles': []
+			}});
+    });
+		err.status.should.equal(401);
+		err.name.should.equal('unauthorized');
+		err.message.should.equal('You are not authorized to access this db.');
+  });
+  it('query', async () => {
+    const userCtx = {
+			name: null,
+			roles: ['member']
+		}
+    const err = await shouldThrowError(async () => {
+      await db.query({map: null}, {userCtx: userCtx});
+    });
+		err.status.should.equal(401);
+
+    const err2 = await shouldThrowError(async () => {
+      await db.query('unexisting-view-func', {userCtx: userCtx});
+    });
+		err2.status.should.equal(404);
+  });
+  it('compact', async () => {
+    //member may not compact
+    const err = await shouldThrowError(async () => {
+      await db.compact({userCtx: {
+				name: 'member',
+				roles: []
+			}});
+    });
+		err.status.should.equal(401);
+		//admin may compact
+		(await db.compact({userCtx: {
+			name: 'admin',
+			roles: []
+		}})).ok.should.be.ok;
+		//server admin (admin party = default) may compact
+		(await db.compact()).ok.should.be.ok;
+  });
+});
+/*
```

Comment:
The functions that still need to be ported from Python -> JS code. 'doc modifications' could be added after ``compact()`` in the same describe() block.